### ISSUE TITLE
Legacy handlebars-helpers usage

### ISSUE THREAD

**Issue by p00h** [assemble/assemble#813]
I've been idle with assemble for a long time.
Good to know you've done that job.
But I wonder how to use legacy `handlebars-helpers` package?
The old style seems not to be the right way.

**Comment by assemblebot** [comment]
@p00h Thanks for the issue! If you're reporting a bug, please be sure to include:

- The version of `assemble` you are using.
- Your assemblefile.js (This can be in a gist)
- The commandline output. (Screenshot or gist is fine)
- What you expected to happen instead.


If this issue is about using assemble with grunt, please open an issue
on the [grunt-assemble](https://github.com/assemble/grunt-assemble/issues/new) repository.

**Comment by p00h** [comment]
I mean we now have in api:
```js
app.engine('hbs', require('engine-handlebars'));
app.helper('helper_name', function(){});
```
But `handlebars-helpers` lib wraps with:
```js
...
Handlebars.registerHelper(name, helper);
```

**Comment by jonschlinkert** [comment]
try using the dev branch on handlebars-helpers. also take a look at https://github.com/jonschlinkert/template-helpers for generic, non-engine-specific helpers that can be used with assemble

**Comment by p00h** [comment]
@jonschlinkert thank you for your quick reply!

**Comment by p00h** [comment]
Put this at your `assemblefile` to have an ability to import old style `handlebars-helpers`:
```js
// Im considering you have assemble initiated at app variable
app.engine('hbs', require('engine-handlebars'));
var handlebars_mock = (function(){
    var myapp = app; 
    return {
        registerHelper: function(name, code){
            myapp.helper(name, code);
        },
        SafeString: function(value){
            return value;
        }
    }
})();

app.helpers(require('handlebars-helpers').register(handlebars_mock, {}, {}));
```

**Comment by jonschlinkert** [comment]
thanks @p00h, good idea!

**Comment by jpgcode** [comment]
Hi @jonschlinkert I'm using 0.9 now and I would not want to use the legacy version of handlebars-helpers because the grunt dependency. I tested downloading the version from dev branch which looks more updated using gulp. It works perfect for me. My question is:

What is the best way to include the dev version in my project? I mean, in the package.json I think I can't reference the dev branch, and I would not want to do it since something could change drastically and break my code. Would you recommend to fork a version and include in my package.json pointing to my repo so when I run npm install it pull it from there?

Thanks in advance.

**Comment by doowb** [comment]
@jpgcode I'm using the dev branch directly [in assemblebot](https://github.com/assemble/assemblebot/blob/master/package.json#L32) but you can point to a specific commit hash on github and `npm` will install it correctly for you. We're working on getting `handlebars-helpers` updated so you won't have to do.

And [here](https://github.com/assemble/assemblebot/blob/master/index.js#L55) is where I'm registering the helpers with assemble. The `utils.helpers()` is equivalent to `require('handlebars-helpers')()` in this case.

> something could change drastically 

This is always a concern, but @jonschlinkert has the code pretty stable now and we're mainly working on the documentation pieces.

Hope this helps.

**Comment by jonschlinkert** [comment]
I'll try my best to publish the dev branch this weekend

**Comment by jpgcode** [comment]
Thank you so much guys! @jonschlinkert  and @doowb your work is REALLY great! Keep up with the good work!

**Comment by jonschlinkert** [comment]
hey thank you! glad to be of help

**Comment by jpgcode** [comment]
Hi @doowb I have noticed something else interesting. If I import the helpers as follows:

Package.json:
`"handlebars-helpers": "github:assemble/handlebars-helpers#dev"`

gulpfile.js:
`var helpers = require('handlebars-helpers')();`

then to register the helpers:
app.helpers(helpers);

For some reason the first time gulp runs it works. However when I save the .hbs file, and assemble task runs again... I get the following as output:
`{$ASYNCID$3$16$}`

It looks related with async helpers.

In order to make it work I had to do it as the example you added above:

The require:
`var helpers = require('handlebars-helpers', 'helpers');`

and register the helpers in assemble as follows:
`app.helpers(helpers());`

I'm not sure why that happens, but using the syntax of the 2nd example works perfectly :). I just wanted to raise it here in case someone else have the same issue or the documentation needs somehow be updated.
